### ISSUE TITLE
Broken on latest Pulumi: does not implement `GetMapping`

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi-go-provider#95]
```
(cd provider && go build -o /Users/evanboyle/go/src/github.com/pulumi/pulumi-auto-deploy/bin/pulumi-resource-auto-deploy -ldflags "-X github.com/pulumi/pulumi-auto-deploy/provider/cmd/main.Version=0.0.1-alpha.1686591813+ff0125db.dirty" github.com/pulumi/pulumi-auto-deploy/provider/cmd/pulumi-resource-auto-deploy)
# github.com/pulumi/pulumi-go-provider
../../../../../pkg/mod/github.com/pulumi/pulumi-go-provider@v0.8.0/provider.go:444:10: cannot use &provider{…} (value of type *provider) as pulumirpc.ResourceProviderServer value in return statement: *provider does not implement pulumirpc.ResourceProviderServer (missing method GetMapping)
```

**Comment by EvanBoyle** [comment]
Changing all go.mods to include:


```
require (
	github.com/blang/semver v3.5.1+incompatible
	github.com/pulumi/pulumi/sdk/v3 v3.59.0
)
```

seemed to get me unblocked, but looks like this package needs to be updated to support the newly added provider methods.

**Comment by iwahbe** [comment]
If you're building a binary, you can also use `replace` to force the upgrade for the library. #96 will fix the issue.

**Comment by iwahbe** [comment]
@EvanBoyle Can you clarify what you mean when you say `all go.mods`? 

Can you confirm that #96 compiles cleanly for you?

We (me and @AaronFriel) were having trouble understanding how the error you mentioned is possible, since `provider` embeds a struct that implements `pulumirpc.ResourceProviderServer`.

**Comment by EvanBoyle** [comment]
@iwahbe I did a little bit more investigation and was able to repro this error, but the root cause is slightly different. The boilerplate references a very old version of the go provider SDK from sept 2022 (v0.8.0):

https://github.com/pulumi/pulumi-provider-boilerplate/blob/main/provider/go.mod#L5

Upgrading my go.mod to v0.9.0 gets rid of the error. I did notice that this repo hasn't gotten a new tagged release in 6 months, do we do that at some sort of regular cadence? 

Anyways apologies for the red herring, but would be good think about how we can keep all of these dependencies up to date across boilerplate and other repos.
